test/test_functionalization.py

```diff
@@ -57,40 +64,41 @@ class TestFunctionalization(TestCase):
 
     crossref = False
 
-    def get_logs(self, func, inpt, *, reapply_views=False, run_reinplace=False):
-        inpt_clone = inpt.clone()
-        traced_f = make_fx(_functionalize(func, reapply_views=reapply_views, crossref=self.crossref))(inpt)
+    def get_logs(self, func, inpts, *, reapply_views=False, run_reinplace=False):
+        if not isinstance(inpts, tuple) and not isinstance(inpts, list):
+            inpts = [inpts]
+        inpts_clone = [inpt.clone() for inpt in inpts]
+        traced_f = make_fx(_functionalize(func, reapply_views=reapply_views, crossref=self.crossref))(*inpts)
         if run_reinplace:
-            traced_f = reinplace(traced_f, inpt_clone)
+            traced_f = reinplace(traced_f, *inpts_clone)
         return traced_f.code
 
-    def assert_functionalization(self, func, inpt, *, reapply_views=False, mutated_input_metadata=False):
-        input_clone = inpt.clone()
-        input_clone2 = inpt.clone()
-        input_clone3 = inpt.clone()
+    def assert_functionalization(self, func, inpts, *, reapply_views=False, mutated_input_metadata=False):
+        if not isinstance(inpts, list) and not isinstance(inpts, tuple):
+            inpts = [inpts]
+        clones1 = [inpt.clone() for inpt in inpts]
+        clones2 = [inpt.clone() for inpt in inpts]
+        clones3 = [inpt.clone() for inpt in inpts]
```

Comment:
Okay consider instead we did:
`def assert_functionalization(self, func, *inpts, reapply_views=False, mutated_input_metadata=False)`

then you can treat inpts like a list all the time (don't need to check) and for the clone that can have the pytree treatment attached to it (i.e. `tree_map_only(torch.Tensor, torch.clone, inpts)`) so that we can support pytree inputs for all functions

Reply:
Hm, I didn't do that because the `*` seemed to exist already for other reasons and so I didn't want to mess with that. 

Will consider modifying this in a different PR.

Reply:
`*` is a way to enforce that the subsequent args are kwarg-only. Having `*inpts` has the same effect because every positional arg after `self, func` will be considered part of `inpts` so you won't be able to pass in `reapply_views` or `mutated_input_metadata ` as positional args

Reply:
ooooh TIL

Reply:
Realized I had to rebase anyway so just made this change for assert_functionalization and get_logs